### ISSUE TITLE
False positive: let_var_whitespace with Swift 5.1 property wrappers

### ISSUE THREAD

**Issue by schayes04** [realm/SwiftLint#2980]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

I have adopted the new Swift 5.1 property wrapper feature. It seems to cause a false positive with the opt-in rule `let_var_whitespace`.

Relevant file:
```
import Foundation

public enum Defaults {

    @UserDefault("isFirstTimeUser", defaultValue: true)
    public static var isFirstTimeUser: Bool
}
```

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint --config ../.swiftlint.yml 
Loading configuration from '../.swiftlint.yml'
Linting Swift files at paths 
Linting 'Defaults.swift' (1/1)
<path>/Defaults.swift:6:5: warning: Variable Declaration Whitespace Violation: Let and var should be separated from other statements by a blank line. (let_var_whitespace)
Done linting! Found 1 violation, 0 serious in 1 files.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.37.0
* Installation method used (Homebrew, CocoaPods, building from source, etc)? Homebrew
* Paste your configuration file:

```yml
opt_in_rules:
  - let_var_whitespace
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? No
* Which Xcode version are you using? 11.2.1
* Do you have a sample that shows the issue? 
```
echo "public enum Defaults {
@UserDefault("isFirstTimeUser", defaultValue: true)
public static var isFirstTimeUser: Bool
}" | swiftlint lint --no-cache --use-stdin --enable-all-rules
```

**Comment by akwilliamson** [comment]
What is the status of this? Is anybody working on it?

**Comment by jshier** [comment]
I've found that this can be avoided by adding a doc comment above the property wrapper. Still rather annoying.

**Comment by bdrelling** [comment]
Adding a +1 here and really appreciative of the workaround @jshier !

**Comment by Tayphoon** [comment]
+++

**Comment by DimaZava** [comment]
The same

**Comment by fidmor89** [comment]
+1 
Also, another workaround would be to have the property wrapper in the same line as the variable definition, something like: 
`@Inject var foo: Foo`

**Comment by jpsim** [comment]
If anyone here has started working on this and has hit a dead end, I'm happy to help answer any questions you might have.

The [CONTRIBUTING.md](https://github.com/realm/SwiftLint/blob/master/CONTRIBUTING.md) doc should be a good first step in looking into making changes to SwiftLint.

**Comment by Igor-Palaguta** [comment]
+1

**Comment by Tayphoon** [comment]
+

**Comment by MarshalGeazipp** [comment]
I hit this annoying bug today.
What is the status of this? Is anybody working on it?

**Comment by edorphy** [comment]
+1

**Comment by HuseyinVural** [comment]
It causes a lot of problems when using it with SwiftUI.

**Comment by horse304** [comment]
+1

**Comment by isaacressler** [comment]
I recently ran into this, and decided to see if I could at least figure out what is happening. After some investigation today I at least figured out that the reason for a comment (doc or normal) removing the false positive is due to how the end of a skipped token is calculated on [LetVarWhitespaceRule.swift line 198 ](https://github.com/realm/SwiftLint/blob/55b62cccb3a062f824a6a0598b32fb56c39823ff/Source/SwiftLintFramework/Rules/Style/LetVarWhitespaceRule.swift#L198). Not sure if that's intended or not, but I did come up with a change that seems to work without breaking anything (in my project). Changing it to subtract one from the calculated byte offset (e.g. ` token.offset + token.length - 1`) removes the workaround, but is at best a first step.

I'm going to take a stab at this again when I get a chance, but documenting this here so I don't forget/to help if someone else decides to take a stab at it.

**Comment by iressler** [comment]
I recently ran into this, and decided to see if I could at least figure out what is happening. After some investigation today I at least figured out that the reason for a comment (doc or normal) removing the false positive is due to how the end of a skipped token is calculated on [LetVarWhitespaceRule.swift line 198 ](https://github.com/realm/SwiftLint/blob/55b62cccb3a062f824a6a0598b32fb56c39823ff/Source/SwiftLintFramework/Rules/Style/LetVarWhitespaceRule.swift#L198). Not sure if that's intended or not, but I did come up with a change that seems to work without breaking anything (in my project). Changing it to subtract one from the calculated byte offset (e.g. ` token.offset + token.length - 1`) removes the workaround, but is at best a first step.

I'm going to take a stab at this again when I get a chance, but documenting this here so I don't forget/to help if someone else decides to take a stab at it.

**Comment by macintosh-HD** [comment]
What is the current status of this?

**Comment by 417-72KI** [comment]
+1

**Comment by SimplyDanny** [comment]
There is a potential fix in #3943.

**Comment by SimplyDanny** [comment]
Could you please create another issue for your finding, @MrCOBA? I don't think it is related.

**Comment by peterkos** [comment]
I've also run into this issue with a property wrapper before an enum:

```swift
@Namespace var userNamespace
private enum UserID: Hashable {
    case descLabel
    case descToggle
}
```

Putting `// swiftlint:disable:next let_var_whitespace` didn't work on any line around the error 😩

**Comment by SimplyDanny** [comment]
> Putting `// swiftlint:disable:next let_var_whitespace` didn't work on any line around the error 😩

Something seems bogus there indeed. Could you create a separate issue for it, @peterkos?
